### PR TITLE
Right click ammo action

### PR DESCRIPTION
**Pull request by BlackCrystalic**
## About The Pull Request

Am... good actions for save your time
## Why It's Good For The Game

Idk, gamer
## Changelog

:cl:BlackCrystalic
code: fixed ammo interactions, included attacking with right click
/:cl:

### CODE REVIEW

**Comment by ChungusGamer666** [comment]
i fucking love you

**Comment by ProfessorPopoff** [comment]
You can unload as many bullets as it will let you into the stack, but it won't update the stack itself until you put one bullet back into the bullet container. If you test it, you'll see what I mean, because I can't really explain it well. 

Also it'll say you're loading the magazine while taking bullets out.

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 1]
```diff
@@ -117,27 +117,43 @@
...
-				num_loaded++
-			if(!did_load || !multiload)
-				break
-		if(num_loaded)
-			AM.update_ammo_count()
+		if(LAZYACCESS(modifiers, RIGHT_CLICK))
+			num_loaded += get_ammo(AM, replace_spent)
+		else
+			num_loaded += AM.get_ammo(src, replace_spent)
+
```
Comment (ProfessorPopoff) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm:145]: Adding in `AM.update_appearance()` here on line 145 will fix the appearance not updating when bullets are taken out.

File: code/modules/projectiles/boxes_magazines/_box_magazine.dm [hunk 2]
```diff
@@ -117,27 +117,43 @@
...
+	var/num_loaded = 0
+	for(var/obj/item/ammo_casing/AC in stored_ammo)
+		var/did_load = AM.give_round(AC, replace_spent)
+		if(did_load)
+			stored_ammo -= AC
+			num_loaded++
+		if(!did_load || !multiload)
+			break
+	if(num_loaded)
+		update_ammo_count()
```
Comment (ChungusGamer666) [inline code/modules/projectiles/boxes_magazines/_box_magazine.dm:131]: ignore what popoff said and add the update_appearance() here, easy

**Comment by ChungusGamer666** [comment]
ammo stacks have a modified attackby proc, you need to edit that too!

**Comment by ChungusGamer666** [comment]
image

**Comment by ChungusGamer666** [comment]
its still not right thho

File: mojave/code/modules/projectiles/boxes_magazines/ammo_stacking.dm [hunk 3]
```diff
@@ -85,34 +85,6 @@
...
-			user.transferItemToLoc(AC, src, TRUE)
-			num_loaded++
-			AC.update_appearance()
-
-	if(num_loaded)
-		if(!silent)
-			to_chat(user, span_notice("You load [num_loaded] shell\s into \the [src]!"))
-			playsound(src, 'sound/weapons/gun/general/mag_bullet_insert.ogg', 60, TRUE)
-		update_ammo_count()
-
```
Comment (ChungusGamer666) [inline mojave/code/modules/projectiles/boxes_magazines/ammo_stacking.dm:115]: i dont remember why i did this without calling parent but i am 100% sure it was for an important reason

Comment (BlackCrystalic) [reply mojave/code/modules/projectiles/boxes_magazines/ammo_stacking.dm:115]: Already in commit upper

Comment (BlackCrystalic) [reply mojave/code/modules/projectiles/boxes_magazines/ammo_stacking.dm:115]: and that same code, bruh

**Comment by Hekzder** [comment]
Resolved by #2292

**Pull request closed without merging**